tests/main/ubuntu-core-writablepaths/task.yaml

```diff
@@ -0,0 +1,37 @@
+summary: Ensure that the writable paths on the image are correct
+systems: [ubuntu-core-16-64]
+execute: |
+    echo "Ensure everything in writable-paths is actually writable"
+    missing=""
+    broken=""
+    while read -r line; do
+        line=$(echo $line | sed -e '/\s*#.*$/d')
+        if [ -z "$line" ]; then
+            continue;
+        fi
+        # a writable-path may be either a file or a directory
+        # 
+        dir_or_file=$(echo $line|cut -f1 -d' ')
+        if [ ! -e "$dir_or_file" ]; then
+            missing="$dir_or_file\n$missing"
+        elif [ -f "$dir_or_file" ]; then
+            if ! touch "$dir_or_file"; then
+                broken="$dir_or_file\n$broken"
+            fi
+        elif ! touch "$dir_or_file"/random-name-that-I-made-up; then
+            broken="$dir_or_file\n$broken"
+        fi
+        rm -f $dir_or_file/random-name-that-I-made-up
+    done < /etc/system-image/writable-paths
```

Comment:
Can we move this ahead, with `cat /etc/system-image/writable-paths | \`? Having at the end means not quite understanding what we're parsing until we find out the stdin coming.